src/model/test/CoilWaterHeatingAirToWaterHeatPumpVariableSpeed_GTest.cpp

```diff
@@ -0,0 +1,170 @@
+  EXPECT_TRUE(coil.setCrankcaseHeaterCapacity(50));
+  EXPECT_TRUE(coil.setMaximumAmbientTemperatureforCrankcaseHeaterOperation(75));
+  EXPECT_TRUE(coil.setEvaporatorAirTemperatureTypeforCurveObjects("DryBulbTemperature"));
+  auto curve = CurveQuadratic(m);
+  EXPECT_TRUE(coil.setPartLoadFractionCorrelationCurve(curve));
+
+  EXPECT_EQ(2, coil.nominalSpeedLevel());
+  EXPECT_EQ(1800.0, coil.ratedWaterHeatingCapacity());
+  EXPECT_EQ(20.0, coil.ratedEvaporatorInletAirDryBulbTemperature());
+  EXPECT_EQ(15.0, coil.ratedEvaporatorInletAirWetBulbTemperature());
+  EXPECT_EQ(60.0, coil.ratedCondenserInletWaterTemperature());
+  ASSERT_TRUE(coil.ratedEvaporatorAirFlowRate());
+  EXPECT_EQ(100.0, coil.ratedEvaporatorAirFlowRate().get());
+  EXPECT_FALSE(coil.isRatedEvaporatorAirFlowRateAutocalculated());
+  ASSERT_TRUE(coil.ratedCondenserWaterFlowRate());
+  EXPECT_EQ(200.0, coil.ratedCondenserWaterFlowRate());
+  EXPECT_FALSE(coil.isRatedCondenserWaterFlowRateAutocalculated());
+  EXPECT_EQ("No", coil.evaporatorFanPowerIncludedinRatedCOP());
+  EXPECT_EQ("Yes", coil.condenserPumpPowerIncludedinRatedCOP());
+  EXPECT_EQ("Yes", coil.condenserPumpHeatIncludedinRatedHeatingCapacityandRatedCOP());
+  EXPECT_EQ(0.3, coil.fractionofCondenserPumpHeattoWater());
+  EXPECT_EQ(50, coil.crankcaseHeaterCapacity());
+  EXPECT_EQ(75, coil.maximumAmbientTemperatureforCrankcaseHeaterOperation());
+  EXPECT_EQ("DryBulbTemperature", coil.evaporatorAirTemperatureTypeforCurveObjects());
+  EXPECT_TRUE(coil.partLoadFractionCorrelationCurve().optionalCast<CurveQuadratic>());
+
+  coil.autocalculateRatedEvaporatorAirFlowRate();
+  coil.autocalculateRatedCondenserWaterFlowRate();
+
+  EXPECT_FALSE(coil.ratedEvaporatorAirFlowRate());
+  EXPECT_TRUE(coil.isRatedEvaporatorAirFlowRateAutocalculated());
+  EXPECT_FALSE(coil.ratedCondenserWaterFlowRate());
+  EXPECT_TRUE(coil.isRatedCondenserWaterFlowRateAutocalculated());
+}
+
+TEST_F(ModelFixture, CoilWaterHeatingAirToWaterHeatPumpVariableSpeed_Clone) {
+  Model m;
+
+  CoilWaterHeatingAirToWaterHeatPumpVariableSpeed coil(m);
+
+  coil.setRatedWaterHeatingCapacity(1900.0);
+
+  CoilWaterHeatingAirToWaterHeatPumpVariableSpeed coilClone = coil.clone(m).cast<CoilWaterHeatingAirToWaterHeatPumpVariableSpeed>();
+  ASSERT_EQ(1900.0, coilClone.ratedWaterHeatingCapacity());
+
+  Model m2;
+  ;
+  CoilWaterHeatingAirToWaterHeatPumpVariableSpeed coilClone2 = coil.clone(m2).cast<CoilWaterHeatingAirToWaterHeatPumpVariableSpeed>();
+  ASSERT_EQ(1900.0, coilClone2.ratedWaterHeatingCapacity());
+}
+
+TEST_F(ModelFixture, CoilWaterHeatingAirToWaterHeatPumpVariableSpeed_Remove) {
+  Model m;
+  auto count = m.modelObjects().size();
+
+  CoilWaterHeatingAirToWaterHeatPumpVariableSpeed coil(m);
+  CoilWaterHeatingAirToWaterHeatPumpVariableSpeedSpeedData speed1(m);
+  coil.addSpeed(speed1);
+  coil.remove();
+
+  auto curves = m.getModelObjects<model::Curve>();
+
+  EXPECT_EQ(count, m.modelObjects().size() - curves.size());
+}
```

Comment:
Please also test what happens to the SpeedDataList and the SpeedDatas themselves when cloning / removing.

Reply:
Done in f0409908eeb8aa66a715847f96a9b0f6f298e25c